sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/JacksonVersionMismatchError.java

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.jackson;
+
+final class JacksonVersionMismatchError extends Error {
```

Comment:
note we always throw it instead of another Error (`LinkageError` such as `NoClassDefFoundError`).  This is an unrecoverable condition and we shouldn't start converting it to Exception.

Reply:
rethrow linkageerror instead